src/components/SharedComponents/TaxonResult.tsx

```diff
@@ -75,23 +84,56 @@ const TaxonResult = ( {
   // network requests for useTaxon instead of making individual API calls.
   // right now, this fetches a single taxon at a time on AI camera &
   // a short list of taxa from offline Suggestions
-  const { taxon: localTaxon } = useTaxon( taxonProp, fetchRemote );
+  const { taxon: localTaxon } = useTaxon( taxonProp, fetchRemote ) as { taxon: RealmTaxon };
   const usableTaxon = fromLocal
     ? localTaxon
     : taxonProp;
-  // useTaxon could return null, and it's at least remotely possible taxonProp is null
-  if ( !usableTaxon ) return null;
-
-  const taxonImage = { uri: usableTaxon?.default_photo?.url };
   const accessibleName = accessibleTaxonName( usableTaxon, currentUser, t );
-
-  const navToTaxonDetails = ( ) => {
+  const navToTaxonDetails = React.useCallback( ( ) => {
     navigation.push( "TaxonDetails", {
       id: usableTaxon?.id,
       hideNavButtons,
       lastScreen,
       vision
     } );
+  }, [
+    hideNavButtons,
+    lastScreen,
+    navigation,
+    usableTaxon?.id,
+    vision
+  ] );
+  const TaxonResultMain = React.useCallback( ( props: TaxonResultMainProps ) => (
+    unpressable
+      // eslint-disable-next-line react/jsx-props-no-spreading
+      ? <View {...props}>{ props.children }</View>
+      : (
+        <Pressable
+          // eslint-disable-next-line react/jsx-props-no-spreading
+          {...props}
+          onPress={handleTaxonOrEditPress || navToTaxonDetails}
+          accessible
+          accessibilityRole="link"
+          accessibilityLabel={accessibleName}
+          accessibilityHint={t( "Navigates-to-taxon-details" )}
+        >
+          { props.children }
+        </Pressable>
+      )
+  ), [
+    accessibleName,
+    handleTaxonOrEditPress,
+    navToTaxonDetails,
+    t,
+    unpressable
+  ] );
+
+  // useTaxon could return null, and it's at least remotely possible taxonProp is null
+  if ( !usableTaxon ) return null;
+
+  const taxonImage = {
+    uri: ( usableTaxon as ApiTaxon )?.default_photo?.url
```

Comment:
Again, this was for TypeScript. The `taxon` prop could be from the API or Realm, and `useTaxon()` will return a Realm object, so to satisfy TypeScript we need these casts, and since there is no runtime type checking in TypeScript since it all compiles to JS, we have to do this duck type check.

Reply:
Let me put it another way, since I didn't notice the more appropriate line when I made this comment: L40 says `taxon: RealmTaxon | ApiTaxon;` and I expected (perhaps from too many years in object-oriented code) that it would say `taxon: Taxon;` so that this component wouldn't have (or need!) knowledge of its data's source. I'm curious to learn if there's a different pattern here that I should learn.

Reply:
Ah, yeah, it should be a single class that gets instantiated with data from the API or the database and presents a single interface, but it's not. Since we have only partially adopted TypeScript, a lot of what we're doing with it is just trying to work around some of the JS problems we have, one of which is throwing around POJOs from the API and Realm objects as if they were equivalent, but they don't even present consistent camelcase / snakecase members. Using TypeScript to at least describe the madness is an improvement, but you're right, the correct solution would be a single class.